src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/GeneratedDocumentSynchronizer.cs

```diff
@@ -9,20 +9,14 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer;
 
 internal class GeneratedDocumentSynchronizer(
     IGeneratedDocumentPublisher publisher,
-    IDocumentVersionCache documentVersionCache,
     LanguageServerFeatureOptions languageServerFeatureOptions) : IDocumentProcessedListener
 {
     private readonly IGeneratedDocumentPublisher _publisher = publisher;
-    private readonly IDocumentVersionCache _documentVersionCache = documentVersionCache;
     private readonly LanguageServerFeatureOptions _languageServerFeatureOptions = languageServerFeatureOptions;
 
     public void DocumentProcessed(RazorCodeDocument codeDocument, IDocumentSnapshot document)
     {
-        if (!_documentVersionCache.TryGetDocumentVersion(document, out var hostDocumentVersion))
-        {
-            // Could not resolve document version
-            return;
-        }
+        var hostDocumentVersion = document.Version;
```

Comment:
We should perhaps check here if the document is open, from project state manager, unless `UpdateBuffersForClosedDocuments` is set.